src/interactivity/win32/UiaTextRange.cpp

```diff
@@ -466,7 +466,10 @@ IFACEMETHODIMP UiaTextRange::Compare(_In_opt_ ITextRangeProvider* pRange, _Out_
     }
     // tracing
     ApiMsgCompare apiMsg;
-    apiMsg.OtherId = other->GetId();
+    if (other)
+    {
+        apiMsg.OtherId = other->GetId();
+    }

```

Comment:
if `other` isn't there can this set `apiMsg.OtherId` to 0 (preferably with 0 being defined as some invalid id constant)? The static `UiaTextRange::id` would also need to be changed to start at 1 instead of 0, but that way we can still send the tracing message and detect when we got a bad api call.

Reply:
Where and how (static, macro, etc) to define the constant?

Reply:
`UiaTextRange::id` is defined near the top of UiaTextRange.hpp and it's initialized to 0 near the top of UiaTextRange.cpp. For the invalid id constant a `#define` in UiaTextRange.hpp would be fine.